nifi-nar-bundles/nifi-standard-bundle/nifi-standard-parameter-providers/src/main/java/org/apache/nifi/parameter/DatabaseParameterProvider.java

```diff
@@ -0,0 +1,254 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+@Tags({"database", "dbcp", "sql"})
+@CapabilityDescription("Fetches parameters from database tables")
+
+public class DatabaseParameterProvider extends AbstractParameterProvider implements VerifiableParameterProvider {
+
+    protected final static Map<String, DatabaseAdapter> dbAdapters = new HashMap<>();
+
+    public static final PropertyDescriptor DB_TYPE;
+
+    static {
+        // Load the DatabaseAdapters
+        ArrayList<AllowableValue> dbAdapterValues = new ArrayList<>();
+        ServiceLoader<DatabaseAdapter> dbAdapterLoader = ServiceLoader.load(DatabaseAdapter.class);
+        dbAdapterLoader.forEach(it -> {
+            dbAdapters.put(it.getName(), it);
+            dbAdapterValues.add(new AllowableValue(it.getName(), it.getName(), it.getDescription()));
+        });
+
+        DB_TYPE = new PropertyDescriptor.Builder()
+                .name("db-type")
+                .displayName("Database Type")
+                .description("The type/flavor of database, used for generating database-specific code. In many cases the Generic type "
+                        + "should suffice, but some databases (such as Oracle) require custom SQL clauses. ")
+                .allowableValues(dbAdapterValues.toArray(new AllowableValue[dbAdapterValues.size()]))
+                .defaultValue("Generic")
+                .required(true)
+                .build();
+    }
+
+    static AllowableValue GROUPING_BY_COLUMN = new AllowableValue("grouping-by-column", "Column",
+            "A single table is partitioned by the 'Parameter Group Name Column'.  All rows with the same value in this column will " +
+                    "map to a group of the same name.");
+    static AllowableValue GROUPING_BY_TABLE_NAME = new AllowableValue("grouping-by-table-name", "Table Name",
+            "An entire table maps to a Parameter Group.  The group name will be the table name.");
+
+    public static final PropertyDescriptor DBCP_SERVICE = new PropertyDescriptor.Builder()
+            .name("dbcp-service")
+            .displayName("Database Connection Pooling Service")
+            .description("The Controller Service that is used to obtain a connection to the database.")
+            .required(true)
+            .identifiesControllerService(DBCPService.class)
+            .build();
+
+    public static final PropertyDescriptor PARAMETER_GROUPING_STRATEGY = new PropertyDescriptor.Builder()
+            .name("parameter-grouping-strategy")
+            .displayName("Parameter Grouping Strategy")
+            .description("The strategy used to group parameters.")
+            .required(true)
+            .allowableValues(new AllowableValue[] { GROUPING_BY_COLUMN, GROUPING_BY_TABLE_NAME })
```

Comment:
Nitpick, my IDE said you can just use the varargs signature rather than creating an array :)